yt_dlp/YoutubeDL.py

```diff
@@ -603,9 +609,23 @@ def __init__(self, params=None, auto_init=True):
         except Exception as e:
             self.write_debug(f'Failed to enable VT mode: {e}')
 
+        if self.params.get('no_color'):
+            if self.params.get('color') is not None:
+                self.report_warning('Using')
+            self.params['color'] = 'no_color'
+
+        term_allow_color = os.environ.get('TERM', '').lower() != 'dumb'
+
+        def process_color_policy(stream):
+            stream_name = {sys.stdout: 'stdout', sys.stderr: 'stderr'}.get(stream)
+            policy = traverse_obj(self.params, ('color', (stream_name, None)), get_all=False)
+            if policy not in (True, False, 'no_color'):
+                return term_allow_color and supports_terminal_sequences(stream)
+            return policy
```

Comment:
In retrospect, True/False/None is a bad idea.
```suggestion
            if policy in ('auto', None):
                return term_allow_color and supports_terminal_sequences(stream)
            assert policy in ('always', 'never', 'no_color')
            return {'always': True, 'never': False}.get(policy, policy)
```